src/runtime/threading_backend.cc

```diff
@@ -134,21 +134,28 @@ class ThreadGroup::Impl {
 #endif
     }
     if (exclude_worker0) {  // bind the master thread to core 0
-      cpu_set_t cpuset;
-      CPU_ZERO(&cpuset);
       if (reverse) {
-        CPU_SET(sorted_order_[sorted_order_.size() - 1], &cpuset);
+        core0_id_ = sorted_order_[sorted_order_.size() - 1];
       } else {
-        CPU_SET(sorted_order_[0], &cpuset);
+        core0_id_ = sorted_order_[0];
       }
+     pthread_atfork(nullptr, nullptr, ThreadGroup::Impl::BindMasterThreadToCore0);
+    }
+#endif
+  }
+
+  static void BindMasterThreadToCore0() {
+#if defined(__linux__) || defined(__ANDROID__)
+    cpu_set_t cpuset;
+    CPU_ZERO(&cpuset);
```

Comment:
We do not want to bind the master thread to core0? Instead, we want to avoid such binding and set its affinity to all cores.

something like

```c++
for (int i = 0; i < std::thread::hardware_concurrency(); ++i) {
   CPU_SET(i, &cpuset);
}
```

Reply:
I think the idea here is to bind the master thread of the child process to "core 0" of the parent process at fork, otherwise there is no binding on the master thread. Actually, I think we can just simply never bind the master thread.

Reply:
OK, I was thinking about the other way. 

- the process can bind the master thread if we want(by default we don't have to)
- Release the binding of the threads at fork(by bind to all cores), so the child thread can freely move around using whatever cores they like

Reply:
> OK, I was thinking about the other way.
> 
> * the process can bind the master thread if we want(by default we don't have to)
> * Release the binding of the threads at fork(by bind to all cores), so the child thread can freely move around using whatever cores they like

Makes sense to me.